src/coprocessor/dag/expr/ctx.rs

```diff
@@ -91,62 +91,68 @@ impl EvalConfig {
         }
     }
 
-    pub fn set_ignore_truncate(mut self, new_value: bool) -> Self {
+    pub fn from_flags(flags: u64) -> Self {
+        let mut config = Self::new();
+        config.set_by_flags(flags);
+        config
+    }
+
+    pub fn set_ignore_truncate(&mut self, new_value: bool) -> &mut Self {
```

Comment:
why change move to mutable ref?

Reply:
Because I found it too ugly when used in DAG.

Previous:

```rust
        let mut eval_cfg = EvalConfig::new().set_by_flags(req.get_flags());
        // We respect time zone name first, then offset.
        if req.has_time_zone_name() && !req.get_time_zone_name().is_empty() {
            eval_cfg = box_try!(eval_cfg.set_time_zone_by_name(req.get_time_zone_name()));
        } else if req.has_time_zone_offset() {
            eval_cfg = box_try!(eval_cfg.set_time_zone_by_offset(req.get_time_zone_offset()));
        } else {
            // This should not be reachable. However we will not panic here in case
            // of compatibility issues.
        }
        if req.has_max_warning_count() {
            eval_cfg = eval_cfg.set_max_warning_cnt(req.get_max_warning_count() as usize);
        }
        if req.has_sql_mode() {
            eval_cfg = eval_cfg.set_sql_mode(req.get_sql_mode());
        }
        if req.has_is_strict_sql_mode() {
            eval_cfg = eval_cfg.set_strict_sql_mode(req.get_is_strict_sql_mode());
        }
```

Now:

```rust
        let mut eval_cfg = EvalConfig::from_flags(req.get_flags());
        // We respect time zone name first, then offset.
        if req.has_time_zone_name() && !req.get_time_zone_name().is_empty() {
            box_try!(eval_cfg.set_time_zone_by_name(req.get_time_zone_name()));
        } else if req.has_time_zone_offset() {
            box_try!(eval_cfg.set_time_zone_by_offset(req.get_time_zone_offset()));
        } else {
            // This should not be reachable. However we will not panic here in case
            // of compatibility issues.
        }
        if req.has_max_warning_count() {
            eval_cfg.set_max_warning_cnt(req.get_max_warning_count() as usize);
        }
        if req.has_sql_mode() {
            eval_cfg.set_sql_mode(req.get_sql_mode());
        }
        if req.has_is_strict_sql_mode() {
            eval_cfg.set_strict_sql_mode(req.get_is_strict_sql_mode());
        }
```